code/modules/mob/living/carbon/carbon.dm

```diff
@@ -531,6 +531,17 @@
 		see_invisible = see_override
 
 
+/mob/living/carbon/proc/update_color_tint(tintcolor)
+	switch(tintcolor)
+		if(TINT_COLOR_GREEN)
+			overlay_fullscreen("color_tint", /obj/screen/fullscreen/color_vision/green)
+		if(TINT_COLOR_RED)
+			overlay_fullscreen("color_tint", /obj/screen/fullscreen/color_vision/red)
+		if(TINT_COLOR_BLUE)
+			overlay_fullscreen("color_tint", /obj/screen/fullscreen/color_vision/blue)
+		else
+			clear_fullscreen("color_tint", 0)
+

```

Comment:
I feel this would be better if it just created the overlay and directly set its color to tintcolor rather than needing a new type of object whose only difference is one var